types/index.d.ts

```diff
@@ -253,31 +247,34 @@ declare namespace unified {
    * A pairing of a plugin with its settings
    *
    * @typeParam S Plugin settings
-   * @typeParam S2 Extra plugin settings
    * @typeParam P Processor settings
    */
-  type PluginTuple<S = Settings, S2 = undefined, P = Settings> =
-    | [Plugin<S, undefined, P>, S]
-    | [Plugin<S, S2, P>, S, S2]
+  type PluginTuple<S extends any[] = [Settings?], P = Settings> = [
+    Plugin<S, P>,
+    /**
+     * NOTE: ideally this would be S instead of any[]
+     * As of TypeScript 3.5.2 generic tuples cannot be spread
+     */
+    ...any[]
+  ]
 
   /**
    * A union of the different ways to add plugins to unified
    *
    * @typeParam S Plugin settings
-   * @typeParam S2 Extra plugin settings
    * @typeParam P Processor settings
    */
-  type Pluggable<S = Settings, S2 = undefined, P = Settings> =
-    | Plugin<S, S2, P>
-    | Preset<P>
-    | PluginTuple<S, S2, P>
+  type Pluggable<S extends any[] = [Settings?], P = Settings> =
+    | Plugin<S, P>
+    | Preset<S, P>
+    | PluginTuple<S, P>
 
   /**
    * A list of plugins and presets
    *
    * @typeParam P Processor settings
    */
-  type PluggableList<P = Settings> = Array<Pluggable<any, any, P>>
+  type PluggableList<P = Settings> = Array<Pluggable<[any?], P>>
```

Comment:
a quick worth noting here
this cannot be
```
Array<Pluggable<any[], P>>
```

`any[]` has length `number` (infinity), tuples have a fixed length, which causes inferred `PluggableList`s to fail.
Giving a fixed length `[any?]` works for inference.